### PR TITLE
Update 6 test modules according to #2776

### PR DESCRIPTION
**Pull request by abalakh**
Part of #2776
Updated following CLI tests:
* activationkey
* architecture
* computeresource
* contenthost
* contentviewfilter
* metatest template_methods

### CODE REVIEW

**Comment by abalakh** [comment]
```
nosetests tests/foreman/cli/test_activationkey.py
.SSSSSSSSS...........SSSSSSS.SSSSSS............................SSSS..............................S......SS..............SSS
----------------------------------------------------------------------
Ran 123 tests in 1912.995s

OK (SKIP=32)

nosetests tests/foreman/cli/test_architecture.py
..................................
----------------------------------------------------------------------
Ran 34 tests in 476.633s

OK

nosetests tests/foreman/cli/test_computeresource.py
.SSSSSS..................................
----------------------------------------------------------------------
Ran 41 tests in 428.217s

OK (SKIP=6)

nosetests tests/foreman/cli/test_contenthost.py
........S...........................................
----------------------------------------------------------------------
Ran 52 tests in 4027.063s

OK (SKIP=1)

nosetests tests/foreman/cli/test_contentviewfilter.py
......S.......................S.....................................
----------------------------------------------------------------------
Ran 68 tests in 1847.855s

OK (SKIP=2)
```

**Comment by sthirugn** [comment]
wow  636 additions and 1,266 deletions! It requires a lot of time to review this code line by line.  I looked at part of this PR and it looks great.  I am gonna ack this so jenkins run can validate this more.

ACK

File: tests/foreman/cli/test_computeresource.py [hunk 1]
```diff
@@ -84,23 +78,16 @@ def test_list(self):
...
-        self.assertEquals(result_list.return_code, 0,
-                          "ComputeResource list - exit code")
-        self.assertTrue(len(result_list.stdout) > 0,
-                        "ComputeResource list - stdout has results")
-        stdout = ComputeResource.exists(
-            search=('name', result_create['name'])).stdout
-        self.assertTrue(
-            stdout,
-            "ComputeResource list - exists name")
+        self.assertTrue(comp_res['name'])
```
Comment (omaciel) [inline tests/foreman/cli/test_computeresource.py]: Did you really mean to assert this as a `bool` or was the intention to make sure that `name` matches the string `"ComputeResource create - has name"`?

File: tests/foreman/cli/test_contentviewfilter.py [hunk 2]
```diff
@@ -334,42 +286,31 @@ def test_create_cvf_with_multiple_repo_with_docker(self):
...
+        ContentView.add_repository({
             u'id': self.content_view['id'],
             u'repository-id': docker_repository['id'],
         })
-        self.assertEqual(result.return_code, 0)
-
-        repos = '{0},{1}'.format(self.repo['id'], docker_repository['id'])
-
-        result = ContentView.filter_create({
+        repos = [self.repo['id'], docker_repository['id']]
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviewfilter.py]: This change will pass `repos` as a `list` and not as a `str` (which was the case). Is this intentional?

**Comment by omaciel** [comment]
ACK

Will merge without rebase.

**Comment by abalakh** [comment]
@sthirugn @omaciel it's PR for temporary `issue_2776` branch, it won't affect `master` in any way. When all the CLI tests are ready, it will be merged to master, so no changes on jenkins for now :)
The reason of separate branch is #2776 is backwards incompatible change, so literally _all_ the CLI tests should be updated & merged altogether with #2776.

**Comment by sthirugn** [comment]
ahh I missed that this is going to a specific branch.

**Comment by sthirugn** [comment]
the sooner this work is done and merged to master, it is good for you :)  otherwise you will see a lot of merge conflicts when you merge to master (ddt removal work is going on in master)

**Pull request merged**